src/jbi/whiteboard_actions/default.py

```diff
@@ -6,35 +6,130 @@
+
+        if not linked_issue_key:
+            # noop
+            return {"status": "noop"}
+        # else
+        jira_response = self.jira_client.issue_add_comment(
+            issue_key=linked_issue_key,
+            comment=payload.map_as_jira_comment(),
+        )
+        return {"status": "comment", "jira_response": jira_response}
+
+    def bug_create_or_update(
+        self, payload: BugzillaWebhookRequest
+    ):  # pylint: disable=too-many-locals
+        """Create and link jira issue with bug, or update; rollback if multiple events fire"""
+        bug_obj = payload.bug
+        linked_issue_key = bug_obj.extract_from_see_also()  # type: ignore
+
+        if linked_issue_key:
+            # update
+            fields, comments = payload.map_as_tuple_of_field_dict_and_comments()
+            jira_update_response = self.jira_client.update_issue_field(
+                key=linked_issue_key, fields=fields
+            )
+            # comment
+            jira_comment_responses = []
+            for comment in comments:
+                jira_comment_responses.append(
+                    self.jira_client.issue_add_comment(
+                        issue_key=linked_issue_key, comment=comment
+                    )
+                )
+            return {
+                "status": "update",
+                "jira_update_response": jira_update_response,
+                "jira_comment_responses": jira_comment_responses,
+            }
+        # else: create jira issue
+        fields = {**bug_obj.get_jira_issue_dict(), "key": self.jira_project_key}  # type: ignore
+
+        response = self.jira_client.create_issue(fields=fields)
+
+        # Jira response can be of the form: List or Dictionary
+        if isinstance(response, list):
+            # if a list is returned, get the first item
+            response = response[0]
+
+        if isinstance(response, dict):
+            # if a dict is returned or the first item in a list, confirm there are no errors
+            if any(
+                element in ["errors", "errorMessages"] and response[element]
+                for element in response.keys()
+            ):
+                raise ActionError(f"response contains error: {response}")
+
+        jira_key_in_response = response.get("key")
+        if not payload.bug:
+            raise ActionError("payload does not contain bug data")
+        current_bug_info = self.bugzilla_client.getbug(payload.bug.id)
+        bug_obj = BugzillaBug.parse_obj(current_bug_info.__dict__)
+        jira_key_in_bugzilla = bug_obj.extract_from_see_also()
+        _duplicate_creation_event = (
+            jira_key_in_bugzilla is not None
+            and jira_key_in_response != jira_key_in_bugzilla
+        )
+        if _duplicate_creation_event:
+            response = self.jira_client.delete_issue(
+                issue_id_or_key=jira_key_in_response
+            )
+            return {"response": response}
```

Comment:
Why not `jira_response` here? 

Also, in `comment_create_or_noop()`  you return a  `status` key, and you don't here. It would be good to have some convention between the two, so that `__call__()` has consistent return values